### ISSUE TITLE
Server crashes when generating a new world

### ISSUE THREAD

**Issue by ffjdksl** [DRE2N/DungeonsXL#527]
### Expected behavior
The dungeon map loads and the game should start, the server doesn't crash and everything fine

### Actual behavior
The dungeon map starts loading, doesn't finish and then the server crashes

### Steps to reproduce
1- Keep the server on for like 8 hours (if you start the server and try it right away it doesn't work)
2- Enter a group, enter the dungeon and the server should crash

Actually, this is more like a "support requesting thread" because there's a crash log in the server when all of this happens, but there are many plugin names so I don't really know which one to remove or what to do. I'll pastebin it right now.

https://pastebin.com/7FCYJf3b

Hopefully you can help, dungeons are actually disabled on my server because of this issue, and that's not very good because my players really like the dungeons I make with this plugin

**Comment by Sataniel98** [comment]
Hi,
basically, the log says what's happening:
`If you see a world save or edit, then it means you did far more than your server can handle at once`
`If this is the case, consider increasing timeout-time in spigot.yml but note that this will replace the crash with LARGE lag spikes`

Loading the instance took just way too long. How large is your dungeon and what are your server specs? Especially if you have imported an existing map, it might have many chunks that are copied & loaded but not used. In this case, try chopping off what you don't need - this tool helps you finding the files that represent unnecessary regions: https://dinnerbone.com/minecraft/tools/coordinates/

It could also be that chunk errors are causing delays. I'm no expert if it comes to this, but I'd try copying the buildings of the problematic dungeon into a new map with WorldEdit or a similar tool.

**Comment by ffjdksl** [comment]
Hi, thanks for your quick answer! My dungeon is like 12-16 chunks large (it's like 2 big roads one in top of other). I copy-pasted the structure from another map we made beforehand using world-edit. (We didn't import the map with the built in command). I'm still going to check that tool when I can tho, thanks for the suggestion. My server has 4gb ram and it's hosted in serverminer.
Again, thanks for your quick answer, and hopefully these infos help you giving an exact answer to my problem. See ya!

**Comment by Sataniel98** [comment]
It's hard to tell what's different between the first try to load a dungeon and one that happens a few hours later. Specs and map size don't seem to be out of the ordinary. If your map had been created before 1.13, it will convert the map each time you load it, but even that still shouldn't take up to a minute of time. DXL's own operations seem to work fine before the actual internal world loading is done. Interestingly, the generation settings are collected correctly, but no loading start region percentages show up. The problem seems to occur when the world loading event for plugins to hook into is fired. I had a look at what the source code of Spigot and PEX do at the listed positions but didn't find any heavy operations, like for example UUID lookup requests.

tl;dr; I'm afraid but I don't know what's wrong, and I can hardly ask you to temporarily remove or replace plugins like Essentials or PermissionsEx.

However, I have noticed that some of the code line numbers in the logs didn't match up with the latest versions of the code on their repositories where I looked it up. That usually means you're running a different, possibly outdated version, and although it might not fix your problem, updating can't harm.

**Comment by ffjdksl** [comment]
Don't worry, I found the solution. Aparently PEX was causing the issue for some reason, now we moved to LPerms and the server hasn't had troubles in a couple hours. If this doesn't fix the issue, i'll try updating to a new paperspigot build. Thanks for your help, hopefully you have an amazing day/night
